sql/core/src/test/scala/org/apache/spark/sql/MapStatusEndToEndSuite.scala

```diff
@@ -39,26 +38,38 @@ class MapStatusEndToEndSuite extends SparkFunSuite with SQLTestUtils {
   }
 
   test("Propagate checksum from executor to driver") {
-    assert(spark.sparkContext.conf
-      .get("spark.sql.shuffle.orderIndependentChecksum.enabled") == "true")
-    assert(spark.conf.get("spark.sql.shuffle.orderIndependentChecksum.enabled") == "true")
     assert(spark.sparkContext.conf.get("spark.sql.leafNodeDefaultParallelism") == "5")
     assert(spark.conf.get("spark.sql.leafNodeDefaultParallelism") == "5")
     assert(spark.sparkContext.conf.get("spark.sql.classic.shuffleDependency.fileCleanup.enabled")
       == "false")
     assert(spark.conf.get("spark.sql.classic.shuffleDependency.fileCleanup.enabled") == "false")
 
-    withTable("t") {
-      spark.range(1000).repartition(10).write.mode("overwrite").
-        saveAsTable("t")
-    }
+    var shuffleId = 0
+    Seq(("true", "false"), ("false", "true"), ("true", "true")).foreach {
+      case (orderIndependentChecksumEnabled: String, checksumMismatchFullRetryEnabled: String) =>
+        withSQLConf(
+          "spark.sql.shuffle.orderIndependentChecksum.enabled" -> orderIndependentChecksumEnabled,
```

Comment:
nit: let's not hardcode it, we can reference them by `SQLConf.key_name`

Reply:
Updated.